test/sequential/test-fs-watchfile.js

```diff
@@ -64,23 +64,47 @@ fs.watchFile(enoentFile, {interval: 0}, common.mustCall(function(curr, prev) {
   }
 }, 2));
 
-// Watch events should callback with a filename on supported systems
-if (common.isLinux || common.isOSX || common.isWindows || common.isAix) {
-  const dir = common.tmpDir + '/watch';
 
-  fs.mkdir(dir, common.mustCall(function(err) {
-    if (err) assert.fail(err);
-
-    fs.watch(dir, common.mustCall(function(eventType, filename) {
-      clearInterval(interval);
-      this._handle.close();
-      assert.strictEqual(filename, 'foo.txt');
-    }));
+function WatchTestCase(is, dirName, fileName, field) {
+  Object.defineProperties(this, {
+    dirPath: { get: () => `${common.tmpDir}/${this.dirName}` },
+    filePath: { get: () => `${this.dirPath}/${this.fileName}` }
+  });
+  Object.assign(this, {is, dirName, fileName, field});
+}
+const cases = [
+  // Watch on a directory should callback with a filename on supported systems
+  new WatchTestCase(
+    common.isLinux || common.isOSX || common.isWindows || common.isAix,
+    'watch1',
+    'foo',
+    'filePath'
+  ),
+  // Watch on a file should callback with a filename on supported systems
+  new WatchTestCase(
+    common.isLinux || common.isOSX || common.isWindows || common.isAix,
+    'watch2',
+    'bar',
+    'dirPath'
+  )
+];
+for (const testCase of cases) {
+  if (!testCase.is) continue;
+  fs.mkdirSync(testCase.dirPath);
+  // long content so it's actually flushed.
+  const content1 = Date.now() + testCase.fileName.repeat(Math.random() * 1e5);
```

Comment:
Why `Math.random()`? I am a bit afraid that this reduces test reproducibility. If you use any random values in a test, please log them to stderr, and if there is an actual reason for `Math.random()` please add a comment that describes it – but I’d guess just using `.repease(10000)` would to the job just fine?